Sources/GRPCReflectionService/Server/ReflectionService.swift

```diff
@@ -0,0 +1,176 @@
+  internal struct FileDescriptorProtoData: Sendable {
+    internal let serializedFileDescriptorProto: Data
+    internal let dependencyNames: [String]
+    internal init(serializedFileDescriptorProto: Data, dependenciesNames: [String]) {
+      self.serializedFileDescriptorProto = serializedFileDescriptorProto
+      self.dependencyNames = dependenciesNames
+    }
+  }
+
+  internal let fileDescriptorDataByFilename: [String: FileDescriptorProtoData]
+  internal let serviceNames: [String]
+
+  public init(fileDescriptor: [Google_Protobuf_FileDescriptorProto]) throws {
+    var fileDescriptorDataAux: [String: FileDescriptorProtoData] = [:]
+    var servicesAux: [String] = []
+
+    do {
+      for fileDescriptorProto in fileDescriptor {
+        let protoDataObj = FileDescriptorProtoData(
+          serializedFileDescriptorProto: try fileDescriptorProto.serializedData(),
+          dependenciesNames: fileDescriptorProto.dependency
+        )
+        fileDescriptorDataAux[fileDescriptorProto.name] = protoDataObj
+        servicesAux.append(contentsOf: fileDescriptorProto.service.map { $0.name })
+      }
+    } catch {
+      throw GRPCStatus(
+        code: .invalidArgument,
+        message: "One of the provided file descriptor protos is invalid."
+      )
+    }
+
+    self.fileDescriptorDataByFilename = fileDescriptorDataAux
+    self.serviceNames = servicesAux
+  }
+
+  public func getSerializedFileDescriptorProtos(fileName: String) throws -> [Data] {
+    var toVisit = Deque<String>()
+    var visited = Set<String>()
+    var serializedFileDescriptorProtos: [Data] = []
+    toVisit.append(fileName)
+
+    while !toVisit.isEmpty {
+      let currentFileName = toVisit.popFirst()
+      if let protoData = self.fileDescriptorDataByFilename[currentFileName!] {
+        let serializedFileDescriptorProto = protoData.serializedFileDescriptorProto
+        if !protoData.dependencyNames.isEmpty {
+          toVisit
+            .append(
+              contentsOf: protoData.dependencyNames
+                .filter { name in
+                  return !visited.contains(name)
+                }
+            )
+        }
+        serializedFileDescriptorProtos.append(serializedFileDescriptorProto)
+      } else {
+        throw GRPCStatus(
+          code: .notFound,
+          message: "The provided file or a dependency of the provided file could not be found."
+        )
+      }
+      visited.insert(currentFileName!)
+    }
+    return serializedFileDescriptorProtos
+  }
+}
+
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+public final class ReflectionService: Reflection_ServerReflectionAsyncProvider {
+  private let protoRegistry: ReflectionServiceData
+
+  public init(fileDescriptorProtos: [Google_Protobuf_FileDescriptorProto]) throws {
+    self.protoRegistry = try ReflectionServiceData(
+      fileDescriptor: fileDescriptorProtos
+    )
+  }
+
+  internal func findFileByFileName(
+    fileName: String,
```

Comment:
We can drop the label of `fileName` here because it's evident from the name of the function (otherwise it's `findFileByFileName(fileName: fileName, ...)` which is three `fileName`s in a row...)

```suggestion
  internal func findFileByFileName(
    _ fileName: String,
```